tile_generator/package_flags.py

```diff
@@ -194,12 +194,15 @@ def _apply(self, config_obj, package, release):
         # TODO: Remove the dependency on this in templates
         package['is_app'] = True
 
-        for link_type, link in package.get('consumes', {}).iteritems():
+        for link_name, link in package.get('consumes', {}).iteritems():
             release['consumes'] = release.get('consumes', {})
-            release['consumes'][link_type] = link
-            if 'deployment' in link:
-                release['consumes_cross_deployment'] = release.get('consumes_cross_deployment', {})
-                release['consumes_cross_deployment'][link_type] = link
+            release['consumes'][link_name] = link
+            release['consumes_for_deployment'] = release.get('consumes_for_deployment', {})
+            release['consumes_for_deployment'][link_name] = link.copy()
+            release['consumes_for_deployment'][link_name].pop('type',None)
+            release['consumes_for_deployment'][link_name].pop('optional',None)
```

Comment:
Again, is the `consumes_for_deployment` dict in releases not need these keys?

Reply:
The removals are by design, the "consumes_for_deployment" is a copy prepared for be used in the deployment manifest...